routemaster/state_machine/utils.py

```diff
@@ -70,7 +70,11 @@ def get_current_history(app: App, label: LabelRef) -> History:
         label_name=label.name,
         label_state_machine=label.state_machine,
     ).order_by(
-        History.id.desc(),
+        # Our model type stubs define the `id` attribute as `int`, yet
+        # sqlalchemy actually allows the attribute to be used for ordering like
+        # this; ignore the type check here specifically rather than complicate
+        # our type definitions.
+        History.id.desc(),  # type: ignore
```

Comment:
This feels non-ideal. Is there a way to declare the types of the fields on the _type_ and _instances_ separately?

Reply:
Unfortunately that doesn't seem to be possible at the moment:
```python
from typing import ClassVar


class Thing:
    class_attr: ClassVar[int] = 0

    def __init__(self):
        self.instance_attr: str = ""
        self.class_attr: str = ''


reveal_type(Thing.class_attr)
reveal_type(Thing.instance_attr)  # I'd hope this would error as not existing

t = Thing()
reveal_type(t.class_attr)
reveal_type(t.instance_attr)
```
```
$ mypy thing.py 
thing.py:12: error: Revealed type is 'builtins.int'
thing.py:13: error: Revealed type is 'builtins.str'
thing.py:16: error: Revealed type is 'builtins.int'
thing.py:17: error: Revealed type is 'builtins.str'
```

https://github.com/python/mypy/issues/4886 seems to suggest there might be a solution at some point, however I don't think mypy is there yet.